dialogue-annotations-processor/src/test/resources/com/palantir/myservice/service/MyServiceDialogueServiceFactory.java.generated

```diff
@@ -66,7 +67,7 @@ public final class MyServiceDialogueServiceFactory implements DialogueServiceFac
                     endpointChannelFactory.endpoint(Endpoints.customResponse);
 
             private final Deserializer<Response> customResponseDeserializer = new ErrorHandlingDeserializerFactory(
-                            new MyResponseDeserializer(), new ConjureErrorDecoder())
+                            new ResponseDeserializer(), new ConjureErrorDecoder())
```

Comment:
This is a little bit interesting: It's not entirely clear what behavior we expect in this case -- should a 500 response status still throw an exception, or do we want to get back a `Response` object with code=500?

Reply:
I suppose we'd want to allow custom error decoders, which would make it pretty weird to reason about if the default error decoder differed for `Response` vs other types

Reply:
Good question: I feel like 500s should still be the remit of the error decoder.